File: 02-array-tuples/py/tuples.py
if __name__ == "__main__":
    multi_types = (1, "nce", "upon",'a', True, 3.1416)

    print("first value:", multi_types[0])
    print("last value:", multi_types[-1])

    nested_tuple = ((4,2,6), (-1,3,2), (9,3,5))
    print("nested_tuple: ", nested_tuple)


    assert len(nested_tuple) == 3


    #destructuring / unpacking
    a,b,c,d,e,f = multi_types

    print(a)
    print(b)
    print(c)
    print(d)
    print(e)
    print(f)

    nested2 = ((4, (5, 'Gfg')), (7, (8, 6)))
  
    # printing original list
    print("The original tuple is : " + str(nested2))
  
    # Unpacking nested tuples
    # using list comprehension
    res = [(x, y, z) for x, (y, z) in nested2]
  
    # printing result
    print("The unpacked nested tuple  listis : " + str(res))


   

File: 02-array-tuples/rs/src/tuples.rs
pub fn demo() {
    let multi_types = (1u8, "nce", "upon", 'a', true, 3.1416_f32);

    println!("first value: {}", multi_types.0);
    println!("second value: {}", multi_types.1);

    let nested_tuple = ((4u8, 2i32, 6f32), (-1i32, 3i16, 2u8), (9u8, 3i16, 5f64));
    println!("nested_tuple: {:?}", nested_tuple);

    //destructuring / unpacking
    let (a, b, c, d, e, f) = multi_types;

    println!("{}", a);
    println!("{}", b);
    println!("{}", c);
    println!("{}", d);
    println!("{}", e);
    println!("{}", f);

    let (a2, (b2, c2)) = (0, (1, 2));

    println!("{}", a2);
    println!("{}", b2);
    println!("{}", c2);
}
